src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatCopy = [...fromFormat];
+  const toFormatCopy = [...toFormat];
+
+  const separator = fromFormatCopy.pop();
+  const parts = date.split(separator);
+
+  const dateParts = {};
+
+  fromFormatCopy.forEach((part, index) => {
+    dateParts[part] = parts[index];
+  });
+
+  if (toFormatCopy.includes('YY') && dateParts['YYYY']) {
+    dateParts['YY'] = dateParts['YYYY'].slice(-2);
+  }
+
+  if (toFormatCopy.includes('YYYY') && dateParts['YY']) {
+    dateParts['YYYY'] =
+      dateParts['YY'] < '30' ? '20' + dateParts['YY'] : '19' + dateParts['YY'];
```

Comment:
The logic for determining the century for two-digit years ('YY') assumes that any year less than '30' belongs to the 21st century. This assumption might not be valid in all contexts. Consider revising this logic to ensure it aligns with the expected behavior for your application.